pkg/eventgenerator/http/generator.go

```diff
@@ -0,0 +1,90 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Generator struct {
+	clientset *kubernetes.Clientset
+	config    *rest.Config
+	logger    logger.Logger
+	namespace string
+	podName   string
+}
+
+func NewGenerator(config *rest.Config, log logger.Logger) (*Generator, error) {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %v", err)
+	}
+	return &Generator{
+		clientset: clientset,
+		config:    config,
+		logger:    log,
+		namespace: "default",
+		podName:   fmt.Sprintf("http-eventgen-pod-%d", time.Now().Unix()),
+	}, nil
+}
+
+func (h *Generator) Generate(params map[string]string, countStr, intervalStr string) (error) {
+	url, ok := params["url"]
+	if !ok {
+		return fmt.Errorf("url parameter is required for HTTP event generator")
+	}
+	
+	count, err := strconv.Atoi(countStr)
+	if err != nil {
+		count = -1
+	}
+
+	interval, err := strconv.ParseFloat(intervalStr, 64)
```

Comment:
What about the prefix? [sleep](https://www.gnu.org/software/coreutils/manual/html_node/sleep-invocation.html#sleep-invocation) command supports string: "Xs", "Xm", "Xh", "Xd". Shouldn't we support the same?

Reply:
yes, this would be beneficial. Thanks